packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import moment from 'moment';
+import { withStyles } from '@material-ui/core/styles';
+import DownArrow from '@material-ui/icons/ArrowDropDown';
+import MuiSwitch from '@material-ui/core/Switch';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { DateRangePicker } from '../../components/DateRangePicker';
+import { getDefaultDates, getLimits, GRANULARITY_CONFIG } from '../../utils/periodGranularities';
+import { Content, ContentText } from './Styles';
+import { TUPAIA_ORANGE } from '../../styles';
+import { setDisplayedMapOverlay } from '../../actions';
+
+const Switch = withStyles({
+  switchBase: {
+    '&$checked': {
+      color: TUPAIA_ORANGE,
+    },
+    '&$checked + $track': {
+      backgroundColor: TUPAIA_ORANGE,
+    },
+  },
+  checked: {},
+  track: {},
+})(MuiSwitch);
+
+const ToolsWrapper = styled.div`
+  display: flex;
+`;
+
+const IconWrapper = styled.div`
+  display: flex;
+  border-left: 1px solid rgba(255, 255, 255, 0.5);
+  padding: 8px 0 0 5px;
+`;
+
+const MeasureDatePicker = styled.div`
+  pointer-events: auto;
+  background: #203e5c;
+  padding: 16px 8px;
+  border-bottom-left-radius: ${({ expanded }) => (!expanded ? '5px' : '0')};
+  border-bottom-right-radius: ${({ expanded }) => (!expanded ? '5px' : '0')};
+`;
+
+export const TitleAndDatePickerComponent = ({
+  mapOverlay,
+  onUpdateMeasurePeriod,
+  isExpanded,
+  isMeasureSelected,
+  toggleMeasures,
+  isMeasureLoading,
+  displayedMapOverlays,
+  onSetDisplayedMapOverlay,
+  // TODO: PHX-103: show both date pickers
+  showDatePickerOnlyAfterSecondTitle,
+}) => {
+  const [isSwitchedOn, setIsSwitchedOn] = useState(true);
+  useEffect(() => {
+    setIsSwitchedOn(displayedMapOverlays.includes(mapOverlay?.mapOverlayId));
+  }, [JSON.stringify(displayedMapOverlays)]);
+  const switchOnChange = () => {
+    if (isSwitchedOn) {
+      onSetDisplayedMapOverlay(
+        displayedMapOverlays.filter(mapOverlayId => mapOverlayId !== mapOverlay?.mapOverlayId),
+      );
+    } else {
+      onSetDisplayedMapOverlay([...displayedMapOverlays, mapOverlay?.mapOverlayId]);
+    }
```

Comment:
This makes the logic flow a bit clearer:

```suggestion
    const newDisplayedOverlays = isSwitchedOn
      ? displayedMapOverlays.filter(mapOverlayId => mapOverlayId !== mapOverlay?.mapOverlayId)
      : [...displayedMapOverlays, mapOverlay?.mapOverlayId];
    onSetDisplayedMapOverlay(newDisplayedOverlays);
```

Also, I see we use `mapOverlay?.mapOverlayId` but I wonder what will happened if mapOverlayId is `undefined`? Can we set the displayed overlays to `undefined`, and will this eg be reflected in the url?

Reply:
It could be `{}` in `useEffect()` .

Reply:
Cool, let's not worry about this then if it doesn't affect the user experience (happy to stay as is if front-end works as expected)